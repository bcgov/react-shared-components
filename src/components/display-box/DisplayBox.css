.bcgov-display-box {
  display: flex;
  flex-direction: row;
  padding: 15px;
}

.bcgov-no-padding-bottom {
  padding: 10px;
  padding-bottom: 0px !important;
}

.bcgov-display-icon {
  padding-right: 15px;
}

.bcgov-display-file {
  padding: 0px !important;
  padding-left: 5px !important;
  min-height: 75px;
}

.bcgov-display-file .bcgov-display-icon {
  display: flex;
  align-self: center;
}

.bcgov-display-right-element {
  width: 100%;
  padding-top: 3px;
  display: flex;
  align-self: center;
}

.bcgov-display-left-element {
  padding-left: 0px !important;
}

.bcgov-grey-background {
  background-color: lightgrey;
}

.bcgov-blue-background {
  background-color: rgba(225, 232, 242, 1);
}

.bcgov-border-background {
  border-width: thin;
  border-style: solid;
  border-color: lightgrey;
}

.bcgov-react-icons {
  vertical-align: middle;
}
