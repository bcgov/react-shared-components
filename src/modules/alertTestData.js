import React from "react";
import { Alert } from "../components/alert/Alert";

export function generateAlert(icon, type, styling, element) {
  return <Alert icon={icon} type={type} styling={styling} element={element} />;
}
