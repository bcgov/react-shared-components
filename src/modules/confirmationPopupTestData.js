/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getConfirmationPopupTestData() {
  return { body };
}
