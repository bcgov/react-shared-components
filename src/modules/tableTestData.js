const elements = [
  { name: "BCeID", value: "bobross42" },
  {
    name: "Last Name",
    value: "Ross",
  },
  { name: "First Name", value: "Bob" },
  { name: "Email Address", value: "bob.ross@example.com" },
];

export function getTableElementsTestData() {
  return elements;
}
