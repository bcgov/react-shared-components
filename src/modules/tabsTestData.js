import React from "react";

export const tabs = [
  {
    title: "Tab1",
    content: <p> Tab1 Content... </p>,
  },
  {
    title: "Tab2",
    content: <p> Tab2 Content... </p>,
  },
  {
    title: "Tab3",
    content: <p> Tab3 Content... </p>,
    isDisabled: true,
  },
];
