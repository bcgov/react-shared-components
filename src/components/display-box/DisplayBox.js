import React from "react";
import PropTypes from "prop-types";

import "./DisplayBox.css";

export const DisplayBox = ({ styling, icon, element }) => {
  return (
    <div className={`display-box ${styling}`}>
      {icon && <div className="display-icon">{icon}</div>}
      <div className="display-right-element">{element}</div>
    </div>
  );
};

DisplayBox.propTypes = {
  styling: PropTypes.string,
  icon: PropTypes.element,
  element: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
};

DisplayBox.defaultProps = {
  styling: "",
  icon: null,
};
