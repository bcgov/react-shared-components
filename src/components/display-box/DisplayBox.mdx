import { Meta, Story } from "@storybook/addon-docs/blocks";
import { DisplayBox } from "./DisplayBox";

<Meta title="DisplayBox" component={DisplayBox} />

# Display Box

The display box is used to highlight or group together information with an optional icon.

## API

| Prop    | Type           | Required | Description                                               | Default |
| ------- | -------------- | -------- | --------------------------------------------------------- | ------- |
| styling | string         | false    | css classnames representing the styling you wish to apply | ""      |
| icon    | element        | false    | jsx element representing the icon                         | null    |
| element | string/element | true     | display element jsx for the content of the display box    |         |

<br />

## Display Box (Without Icon)

Used for content with no icon. In this case, rendering a table within the display box.

<Story name="Without Icon" id="displaybox--without-icon"></Story>

```bash
const elements = [
  { name: "BCeID", value: "bobross42" },
  {
    name: "Last Name",
    value: "Ross"
  },
  { name: "First Name", value: "Bob" },
  { name: "Email Address", value: "bob.ross@example.com" }
];

const table = <Table elements={elements} />

<DisplayBox element={table} />;
```

## Display Box (With Side by Side Content)

Used for left and right column content.

<Story
  name="With Side by Side Content"
  id="displaybox--with-side-by-side-content"
></Story>

```bash
const elements = [
  {
    name: <p style={{ width: "80%" }}>This is the content on the left of the display box. Maybe there will be other content on the right too.</p>,
    value: <p style={{ width: "80%" }}>This is the content on the right of the display box. There already is content on the left side of the box.</p>
  },
];

<DisplayBox styling="display-left-element" element={<Table elements={elements} />} />;
```

## Display Box (With Icon)

Used for content with an icon.

<Story name="With Icon" id="displaybox--with-icon"></Story>

```bash
const icon = <MdPerson size={32} />;

<DisplayBox icon={icon} element={table} />;
```

## Display Box (With Blue Background)

Used for content with a gov-themed blue background.

<Story
  name="With Blue Background"
  id="displaybox--with-blue-background"
></Story>

```bash
const icon = <MdPerson size={32} />;

<DisplayBox styling="blue-background" icon={icon} element={table} />
```

## Display Box (With Border)

Used for content with a border around it.

<Story name="With Border" id="displaybox--with-border"></Story>

```bash
<DisplayBox styling="border-background" icon={icon} element={table} />
```

## Display Box (With File Data)

Used to display file information usually with description, icon, and fees.

<Story name="With File Data" id="displaybox--with-file-data"></Story>

```bash
const fileData = [
  {
    name: "Description:",
    value: "some file description",
    isValueBold: true
  },
  {
    name: "Statutory Fee:",
    value: "$10.00",
    isValueBold: true
  }
];

const generateTableData = () => {
  return [
    {
      name: (
        <div style={{ width: "80%" }}>
          <span>file name</span>
        </div>
      ),
      value: (
        <Table elements={fileData} />
      )
    }
  ];
};

const documentIcon = (
  <div style={{ color: "rgb(252, 186, 25)" }}>
    <MdDescription size={32} />
  </div>
);

<DisplayBox
  styling="border-background display-file"
  icon={documentIcon}
  element={<Table elementStyles={{ columnStyle: "bcgov-vertical-middle bcgov-fill-width" }} elements={generateTableData()} />}
/>
```
