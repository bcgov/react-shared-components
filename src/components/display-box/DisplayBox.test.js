import React from "react";
import { MdPerson } from "react-icons/md";

import testBasicSnapshot from "../../TestHelper";
import { DisplayBox } from "./DisplayBox";
import { getTestTable } from "../../modules/displayBoxTestData";

const table = getTestTable();
const icon = <MdPerson size={32} />;

const generateDisplayBox = (hasIcon, styling) => {
  if (!hasIcon) return <DisplayBox element={table} />;
  else return <DisplayBox icon={icon} element={table} styling={styling} />;
};

describe("DisplayBox Component", () => {
  test("Matches the 'without icon' snapshot", () => {
    testBasicSnapshot(generateDisplayBox(false));
  });

  test("Matches the 'with icon' snapshot", () => {
    testBasicSnapshot(generateDisplayBox(true));
  });

  test("Matches the warning snapshot", () => {
    testBasicSnapshot(generateDisplayBox(true, "warning-background"));
  });

  test("Matches the blue-background snapshot", () => {
    testBasicSnapshot(generateDisplayBox(true, "blue-background"));
  });
});
