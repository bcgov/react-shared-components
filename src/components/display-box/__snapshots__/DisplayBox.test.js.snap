// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayBox Component Matches the 'with icon' snapshot 1`] = `
<div
  className="display-box "
>
  <div
    className="display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="display-right-element"
  >
    <div
      className="bcgov-table "
    >
      <b>
        
      </b>
      <div
        className="bcgov-table-body"
      >
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            BCeID
          </div>
          <div>
            bobross42
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Last Name
          </div>
          <div>
            Ross
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            First Name
          </div>
          <div>
            Bob
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Email Address
          </div>
          <div>
            bob.ross@example.com
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DisplayBox Component Matches the 'without icon' snapshot 1`] = `
<div
  className="display-box "
>
  <div
    className="display-right-element"
  >
    <div
      className="bcgov-table "
    >
      <b>
        
      </b>
      <div
        className="bcgov-table-body"
      >
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            BCeID
          </div>
          <div>
            bobross42
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Last Name
          </div>
          <div>
            Ross
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            First Name
          </div>
          <div>
            Bob
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Email Address
          </div>
          <div>
            bob.ross@example.com
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DisplayBox Component Matches the blue-background snapshot 1`] = `
<div
  className="display-box blue-background"
>
  <div
    className="display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="display-right-element"
  >
    <div
      className="bcgov-table "
    >
      <b>
        
      </b>
      <div
        className="bcgov-table-body"
      >
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            BCeID
          </div>
          <div>
            bobross42
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Last Name
          </div>
          <div>
            Ross
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            First Name
          </div>
          <div>
            Bob
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Email Address
          </div>
          <div>
            bob.ross@example.com
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DisplayBox Component Matches the warning snapshot 1`] = `
<div
  className="display-box warning-background"
>
  <div
    className="display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="display-right-element"
  >
    <div
      className="bcgov-table "
    >
      <b>
        
      </b>
      <div
        className="bcgov-table-body"
      >
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            BCeID
          </div>
          <div>
            bobross42
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Last Name
          </div>
          <div>
            Ross
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            First Name
          </div>
          <div>
            Bob
          </div>
        </div>
        <div
          className="bcgov-row "
        >
          <div
            className="bcgov-vertical-middle "
          >
            Email Address
          </div>
          <div>
            bob.ross@example.com
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
